jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -148,9 +153,12 @@ object XGBoost extends Serializable {
    * @param missing the value represented the missing value in the dataset
    * @param featureCol the name of input column, "features" as default value
    * @param labelCol the name of output column, "label" as default value
+   * @param baseMarginCol the name of the column with the initial prediction,
+   *                      "baseMargin as default value
    * @throws ml.dmlc.xgboost4j.java.XGBoostError when the model training is failed
    * @return XGBoostModel when successful training
    */
+  // scalastyle:off
```

Comment:
Personally, I don't think 10 params is a problem, but I don't mind moving `baseMarginCol` along with the other two columns into `params` in the followup PR (moving just `baseMarginCol` feels a bit hacky).

Note however, that his way the API would be harder to discover, since the contract would only be enforced by the documentation (and not code).